vpr/src/pack/cluster.cpp

```diff
@@ -2036,20 +2053,35 @@ static void mark_and_update_partial_gain(const AtomNetId net_id, enum e_gain_upd
 }
 
 /*****************************************/
-static void update_total_gain(float alpha, float beta, bool timing_driven, bool connection_driven, t_pb* pb) {
+static void update_total_gain(float alpha, float beta, bool timing_driven, bool connection_driven, t_pb* pb, AttractionInfo& attraction_groups) {
     /*Updates the total  gain array to reflect the desired tradeoff between*
```

Comment:
write a better overall comment for this routine

Reply:
routine is called when new molecule added to cluster, updates gain for blocks in marked blocks array